enhancements/profile-name-in-rule.md

```diff
@@ -0,0 +1,187 @@
+kind: Rule
+rationale: >-
+  Mounting service account tokens inside pods can provide an avenue for
+  privilege escalation attacks where an attacker is able to compromise a single
+  pod in the cluster.
+title: Restrict Automounting of Service Account Tokens
+id: xccdf_org.ssgproject.content_rule_accounts_restrict_service_account_tokens
+description: >-
+  Service accounts tokens should not be mounted in pods except where the
+  workload running in the pod explicitly needs to communicate with the API
+  server. To ensure pods do not automatically mount tokens, set
+  automountServiceAccountToken to false.
+severity: medium
+apiVersion: compliance.openshift.io/v1alpha1
+```
+
+### User Stories
+
+1. As a cluster administrator, I want to be able to what profiles are using a specific rule
+
+### API Extensions
+
+This enhancement does not require any API changes.
+
+
+### Implementation Details/Notes/Constraints [optional]
+
+What are the caveats to the implementation? What are some important details that
+didn't come across above. Go in to as much detail as necessary here. This might
+be a good place to talk about core concepts and how they relate.
+
+### Risks and Mitigations
+
+What are the risks of this proposal and how do we mitigate. Think broadly. For
+example, consider both security and how this will impact the larger OKD
+ecosystem.
+
+How will security be reviewed and by whom?
+
+How will UX be reviewed and by whom?
+
+Consider including folks that also work outside your immediate sub-project.
+
+## Design Details
+
+### Open Questions [optional]
+
+1. Is `compliance.openshift.io/profiles` a good annotation name?
+2. What happens if users annotate a rule with a Profile?
+   We don't prevent users from doing that, but we should document what we use
+   the annotation and stop users from annotating rules with that annotation key.
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy. Anything
+that would count as tricky in the implementation and anything particularly
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations).
+
+### Upgrade / Downgrade Strategy
+
+We do not expect any upgrade or downgrade issues with this enhancement.
+
+### Version Skew Strategy
+
+Not applicable to this particular enhancement given the Compliance Operator is
+a single installation within a single cluster.
+
+### Operational Aspects of API Extensions
+
+This enhancement does not require any API changes.
+
+#### Failure Modes
+
+Any failure with this enhancement during the profile parsing phase will block
+the user from using the Compliance Operator.
```

Comment:
Just thinking though this a little more.

What would cause the parsing to fail? Would it be possible to have a rule reference a non-existent profile (I didn't think so)?

Would it be possible for a profile to contain a non-existent rule? Since profile rules are a subset of the total rules in a datastream, I wouldn't think so?

Reply:
it can be due to some issues like couldn't list profiles

Reply:
In that case, we're going to have a hard failure and the user will need to remediate. Though - given that it's going to be related to content, that should be found and addressed in CI first (we have coverage for this already).